### ISSUE TITLE
Devnet 

### ISSUE THREAD

**Issue by liamsi** [celestiaorg/celestia-core#491]
## Dev-net

After our MVP we want to be able to deploy a basic dev-net.

### Goals 

The main goal of the dev-net is to test DAS in a much more realistic setting and at the same time test the deployment flow and assumptions about rollups as much as possible.

Hence the minimal features we need before we can launch the devnet are:
- full storage nodes (FS) and DAS light clients
- ability to deploy [optimint](https://github.com/celestiaorg/optimint)/ABCI-based Rollups

#### Stretch goals 

Ideally, we would have fully implemented (state and erasure coding) _fraud proofs_ as well as the propagation part.
Additionally, it would be nice to have Ethermint/some _EVM compatible execution environment_ as a rollup on top of optimint. Also, nice to have are _partial nodes_ but this depends and is blocked on celestiaorg/celestia-specs#200. 

### ADRs and Spec

- [x] Summarize all changes that were already done in lazyledger-core in one ADR #386 
- [x] Define a path for moving the bitswap/IPFS logic into a separate repo
- [x] Describe communication between core/consensus to Celestia node and overall Design of Celestia node (@liamsi)
- [ ] Write networking documentation for networking between Celestia nodes
- [ ] ADR regarding the PartsetHeader (goal: minimize changes to tendermint but still safe) (https://github.com/celestiaorg/celestia-core/issues/511)

The issues below are not directly blocking the dev-net but are critical to increase implementation and specification velocity mid and long-term:

- [x] further improve and codify process between specs and implementation https://github.com/lazyledger/lazyledger-specs/issues/158
- [ ] Critical Component: Consensus
  - [ ] Pseudo-Code for core consensus: https://github.com/lazyledger/lazyledger-specs/issues/176,   
  - [ ] fork-choice & leader selection: https://github.com/lazyledger/lazyledger-specs/issues/148
- [ ] Clarifications and Assumptions around storage nodes: https://github.com/lazyledger/lazyledger-specs/issues/175

### Repositories
- [X] Create Celestia node repo (as laid out in ADR)

### Granular issues
 - [x] Optimint is ready for deploying Rollups (use-case / dev centric [MVP](https://github.com/celestiaorg/optimint/milestone/1))
 - [x] consensus timeouts to achieve 60 sec block time: https://github.com/celestiaorg/lazyledger-core/issues/87
 - [x] https://github.com/celestiaorg/lazyledger-core/issues/436
 - [x] Define and implement DAS light client logic in Celestia node repo
 - [x] #528
 - [x] https://github.com/celestiaorg/celestia-core/issues/509
 - [x] #518 
 - [x] #505
 - [x] #506 
 - [x] https://github.com/lazyledger/lazyledger-core/issues/296
 - [x] Celestia node; for details see: celestiaorg/celestia-node#1 (previous context: #435)

### Chores
- [ ] Code Contribution Guidelines to facilitate community contributions: should be tackled as part of #407 

### Nice to Have 
- [x] serving shares per NID (#221, #422)
- [x] Use a different ipfs-network: https://github.com/celestiaorg/lazyledger-core/issues/154
- [ ] Define data "sharding" for partial nodes: celestiaorg/celestia-specs#200
- [ ] erasure coding fraud proofs #461 
- [ ] (fully) implement state fraud proofs across repos: #459 
- [ ] clarify how much of the custom logic could moved into the app
- [ ] Decide on: git sync #180
- [ ] #511

**Comment by liamsi** [comment]
Congrats everyone 🎉 This issue can be closed. Let's create tracking issues per team and I'll create an overarching project board for the next milestone: testnet.
